workflow-examples/src/main/java/com/redhat/parodos/examples/vmmigration/checker/MigrationStatusWorkFlowChecker.java

```diff
@@ -0,0 +1,90 @@
+import com.redhat.parodos.examples.vmmigration.dto.models.V1beta1MigrationStatusConditions;
+import com.redhat.parodos.examples.vmmigration.utils.Kubernetes;
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.parameter.WorkParameter;
+import com.redhat.parodos.workflow.parameter.WorkParameterType;
+import com.redhat.parodos.workflow.task.checker.BaseWorkFlowCheckerTask;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.JSON;
+import io.kubernetes.client.openapi.apis.CustomObjectsApi;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MigrationStatusWorkFlowChecker extends BaseWorkFlowCheckerTask {
+
+	@Override
+	public WorkReport checkWorkFlowStatus(WorkContext workContext) {
+		log.info("Start MigrationStatusWorkFlowChecker...");
+		try {
+			String migrationName = getRequiredParameterValue(Constants.MIGRATION_NAME_PARAMETER_NAME);
+
+			// Create ApiClient
+			String apiURL = getRequiredParameterValue(Constants.KUBERNETES_API_SERVER_URL);
+			String token = getRequiredParameterValue(Constants.KUBERNETES_TOKEN);
+			String caCert = getOptionalParameterValue(Constants.KUBERNETES_CA_CERT, "");
+			ApiClient client = Kubernetes.buildApiClient(apiURL, token, caCert);
+
+			// Retrieve Migration object
+			CustomObjectsApi coapi = new CustomObjectsApi(client);
+			Object resp = coapi.getNamespacedCustomObject("forklift.konveyor.io", "v1beta1", "demo24", "migrations",
+					migrationName);
+			JSON json = client.getJSON();
+			V1beta1Migration migration = json.deserialize(json.serialize(resp), V1beta1Migration.class);
+
+			V1beta1MigrationStatus status = migration.getStatus();
+			if (status == null) {
+				return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("no status found"));
+			}
+			List<V1beta1MigrationStatusConditions> conditions = status.getConditions();
+			if (conditions == null) {
+				return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("no conditions found"));
+			}
+			for (V1beta1MigrationStatusConditions condition : conditions) {
+				if (condition.getType().equals("Succeeded") && condition.getStatus().equals("True")) {
+					return new DefaultWorkReport(WorkStatus.COMPLETED, workContext);
+				}
+			}
+		}
+		catch (ApiException e) {
+			log.error(e.getResponseBody());
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+		catch (MissingParameterException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+
+		return new DefaultWorkReport(WorkStatus.FAILED, workContext);
+	}
+
+	public List<WorkParameter> getWorkFlowTaskParameters() {
```

Comment:
If I'm not wrong the proper definition of this function should be:

@Override
public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
....
}

can you please double check?